pkg/apiserver/statement/queries.go

```diff
@@ -168,12 +177,21 @@ func (s *Service) queryPlanDetail(
 	query := db.
 		Select(selectStmt).
 		Table(statementsTable).
-		Where("summary_begin_time >= FROM_UNIXTIME(?) AND summary_end_time <= FROM_UNIXTIME(?)", beginTime, endTime).
-		Where("schema_name = ?", schemaName).
-		Where("digest = ?", digest)
+		Where("summary_begin_time >= FROM_UNIXTIME(?) AND summary_end_time <= FROM_UNIXTIME(?)", beginTime, endTime)
+
 	if len(plans) > 0 {
 		query = query.Where("plan_digest in (?)", plans)
 	}
+	if schemaName != "" {
+		query.Where("schema_name = ?", schemaName)
+	}
```

Comment:
If `digest == ""` (if the record is "other record"), then there is no need to check plan digest or schema name I think. This can be a special branch.